static/app/views/performance/browser/webVitals/utils/queries/useInpSpanSamplesWebVitalsQuery.tsx

```diff
@@ -0,0 +1,75 @@
+import type {ReactText} from 'react';
+
+import type {InteractionSpanSampleRowWithScore} from 'sentry/views/performance/browser/webVitals/utils/types';
+import {
+  type Filters,
+  useIndexedSpans,
+} from 'sentry/views/starfish/queries/useIndexedSpans';
+import {SpanIndexedField, SpanMeasurements} from 'sentry/views/starfish/types';
+
+export function useInpSpanSamplesWebVitalsQuery({
+  transaction,
+  limit,
+  enabled,
+  filters = {},
+}: {
+  limit: number;
+  enabled?: boolean;
+  filters?: Filters;
+  transaction?: string;
+}) {
+  const {data, isLoading, ...rest} = useIndexedSpans({
+    filters: {
+      'span.op': 'ui.interaction.click',
+      'measurements.score.weight.inp': '>0',
+      ...(transaction !== undefined
+        ? {[SpanIndexedField.ORIGIN_TRANSACTION]: transaction}
+        : {}),
+      ...filters,
+    },
+    sorts: [{field: SpanIndexedField.TIMESTAMP, kind: 'desc'}],
+    fields: [
+      SpanMeasurements.INP,
+      SpanMeasurements.INP_SCORE,
+      SpanMeasurements.INP_SCORE_WEIGHT,
+      SpanMeasurements.TOTAL_SCORE,
+      SpanIndexedField.ID,
+      SpanIndexedField.TIMESTAMP,
+      SpanIndexedField.PROFILE_ID,
+      SpanIndexedField.REPLAY_ID,
+      SpanIndexedField.USER,
+      SpanIndexedField.ORIGIN_TRANSACTION,
+      SpanIndexedField.PROJECT,
+      SpanIndexedField.BROWSER_NAME,
+      SpanIndexedField.SPAN_SELF_TIME,
+    ],
+    enabled,
+    limit,
+    referrer: 'api.performance.browser.web-vitals.spans',
```

Comment:
I think this needs to be added to the `referrer.py` file and registered as a valid referrer on events, in a separate PR.